### PR TITLE
Add comparisons for lisp strings with native types

### LINKED ISSUE #1291: Can we make string comparisons more ergonomic?
It would be nice to allow comparison of LispStringRef and strings or byte slice via PartialEq.

A potential snag is Emacs is not UTF8 compliant internally. This means using Rust's string support tools will not work without us checking and raising errors.

### PR DESCRIPTION
**Pull request by agraven**
I'd like to have a `PartialEq` implementation that works for both `&str` and `&String`, but `PartialEq<Deref<Target=str>>` doesn't do the trick. Any advice here would be appreciated.

And feel free to bikeshed about the test name.

Closes #1291.

### CODE REVIEW

**Comment by nickdrozd** [comment]
Could you add in at least a few instances where this would be used?

**Comment by agraven** [comment]
Shoot, I can't easily grep my way to finding places where this can be used. I'm gonna do a more thorough search then.

File: rust_src/src/strings.rs [hunk 1]
```diff
@@ -227,6 +227,14 @@ fn str_equality() {
     assert!(!string_equal(string2, string3));
 }
```
Comment (shaleh) [inline rust_src/src/strings.rs]: Let's include some words with letters beyond ASCII in this testing.

**Comment by shaleh** [comment]
Agreed @nickdrozd.

File: rust_src/src/multibyte.rs [hunk 2]
```diff
@@ -193,12 +193,18 @@ impl PartialEq<&[u8]> for LispStringRef {
     }
 }
 
-impl PartialEq<&str> for LispStringRef {
-    fn eq(&self, other: &&str) -> bool {
```
Comment (shaleh) [inline rust_src/src/multibyte.rs]: Yes! Lifetimes finally showing up. Nice.

**Pull request merged**